apps/communications/contacts/js/activities_vcard.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+/* global ActivityHandler, LazyLoader, VCardReader, Contacts */
+
+ActivityHandler.vCardActivityHandler = {
+  handle: function(activity, handler) {
+    const DEPENDENCIES = '/shared/js/contacts/import/utilities/vcard_reader.js';
+    LazyLoader.load(DEPENDENCIES, function() {
+      getvCardReader(activity.source.data.blob).then(readvCard);
+    });
+
+    function getvCardReader(blob) {
+      return new Promise((resolve) => {
+        var fileReader = new FileReader();
+        fileReader.readAsText(blob);
+        fileReader.onloadend = function() {
+          resolve(new VCardReader(fileReader.result));
+        };
```

Comment:
you need to deal with the error as well an reject the promise and the caller should handle the error properly